include/lsst/meas/base/FlagHandler.h

```diff
@@ -137,73 +208,123 @@ class FlagHandler {
+            return record.get(_vector[i].second);
+        }
+        throw FatalAlgorithmError("No legal FlagHandler entry number " + std::to_string(i));
+    }
+    /**
+     *  Return the value of the flag field with the given flag name
+     */
+    bool getValue(afw::table::BaseRecord const & record, std::string flagName) const {
+        for (std::size_t i = 0; i < _vector.size(); i++) {
+            if (_vector[i].first == flagName && _vector[i].second.isValid()) {
+                return record.get(_vector[i].second);
+            }
+        }
+        throw FatalAlgorithmError("No FlagHandler entry for " + flagName);
     }
-
     /**
-     *  Set the flag field corresponding to the given enum value.
+     *  Set the flag field corresponding to the given flag index.
      */
     void setValue(afw::table::BaseRecord & record, std::size_t i, bool value) const {
-        assert(_vector.size() > i);  // Flag 'i' needs to be available
-        record.set(_vector[i].second, value);
+        if (i < _vector.size() && _vector[i].second.isValid()) {
+            record.set(_vector[i].second, value);
+            return;
+        }
+        throw FatalAlgorithmError("No legal FlagHandler entry number " + std::to_string(i));
+    }
+    /**
+     *  Set the flag field corresponding to the given FlagDefinition Address.
+     */
+    /**
+     *  Set the flag field corresponding to the given flag name.
+     */
+    void setValue(afw::table::BaseRecord & record, std::string flagName, bool value) const {
+        for (std::size_t i = 0; i < _vector.size(); i++) {
+            if (_vector[i].first == flagName && _vector[i].second.isValid()) {
+                record.set(_vector[i].second, value);
+                return;
+            }
+        }
+        throw FatalAlgorithmError("No FlagHandler entry for " + flagName);
+    }
+    /**
+     *  Get the index of the General Failure flag, if one is defined.  This flag is defined
+     *  by most algorithms, and if defined, is set whenever an error is caught by the FlagHandler.
+     *  If no General Failure flag is defined, this routine will return FlagDefinition::number_undefined 
+     */
+    std::size_t getFailureFlagNumber() const {
+        return failureFlagNumber;
     }
-
     /**
      *  Handle an expected or unexpected Exception thrown by a measurement algorithm.
      *
      *  If the exception is expected, it should inherit from MeasurementError and can be passed here;
      *  this allows handleFailure to extract the failure mode enum value from the exception and set
      *  the corresponding flag.  The general failure flag will be set regardless of whether the "error"
-     *  argument is null.
+     *  argument is NULL (which happens when an unexpected error occurs).
      */
     void handleFailure(afw::table::BaseRecord & record, MeasurementError const * error=NULL) const;
 
+    std::size_t failureFlagNumber;
 private:
-
```

Comment:
removal of blank lines should be put in a separate commit.  There are a couple of other examples of this below.